docs/scaling/OVERVIEW.md

```diff
@@ -0,0 +1,41 @@
+# Autoscaling
+
+There are three main components to the autoscaling system; autoscaler, activator and queue-proxy.
+
+## Queue-proxy
+Whenever a Serverless service is deployed by the user, the queue-proxy sidecar container is injected with it. This queue proxy does the following:
+ - Any request coming to the application user-container will go to the queue-proxy and then be routed to the user-container
+ - The readiness and liveness probes defined by the user will be replaced with queue-proxy probes externally to the pod while the user-containers probe endpoints will only be accessed by the proxy.
+ - Makes sure that no more than the ‘defined container concurrency’ requests reach the application's instance at once by queueing other requests locally
+ - Collects metrics about the requests load the application user container and exposes that on a separate port
+
+## Autoscaler
+The Autoscaler is responsible for collecting metrics from the queue-proxy of the different application instances and comes up with the desired the number of pods an application needs using the basic calculation: 
+```
+want = concurrencyInSystem/targetConcurrencyPerInstance
+```
+about the calculations:
+ - Requests stats are stored in a ring buffer indexed by timeToIndex() % len(elements in ring). Each element represents a certain granularity of time, and the total represented duration adds up to a window length of time.
+ - With each new star, a value is added with an associated time to the correct element. If this record would introduce a gap in the data, any intervening times between the last write and this one will be recorded as zero. If an entire window length has expired without data, the firstWrite time is reset, meaning the WindowAverage will be of a partial window until enough data is received to fill it again.
```

Comment:
"With each new star..."  is "star" a typo?

Reply:
it was stat :D will change it